src/reconcile/rec_write.c

```diff
@@ -284,7 +284,7 @@ __reconcile(WT_SESSION_IMPL *session, WT_REF *ref, WT_SALVAGE_COOKIE *salvage, u
                 WT_TRET(session->block_manager_cleanup(session));
         }
 
-        __rec_destroy_session(session);
+        WT_TRET(__rec_destroy_session(session));
```

Comment:
This would also require fixing:
```
if (ret == 0 && (ret = __rec_destroy_session(session)  != 0) {
   goto panic;
} else
     WT_TRET(session->block_manager_cleanup(session));
```